File: personal-assistant-agent/src/utils/tokenManager.ts
import fs from 'fs';
import path from 'path';

const tokenPath = path.join(__dirname, 'token.json');

export const saveToken = (token: object) => {
    fs.writeFileSync(tokenPath, JSON.stringify(token));
};

export const loadToken = () => {
    if (fs.existsSync(tokenPath)) {
        const tokenData = fs.readFileSync(tokenPath, 'utf-8');
        return JSON.parse(tokenData);
    }
    return null;
};

File: personal-assistant-agent/src/utils/token_manager.py
import json
import os
from typing import Any, Optional


def _token_path() -> str:
    return os.environ.get("TOKEN_PATH", "token.json")


def save_token(token: object) -> None:
    with open(_token_path(), "w", encoding="utf-8") as f:
        json.dump(token, f)


def load_token() -> Optional[Any]:
    path = _token_path()
    if os.path.exists(path):
        with open(path, encoding="utf-8") as f:
            return json.load(f)
    return None
